crates/router_derive/src/macros/api_error/helpers.rs

```diff
@@ -260,32 +260,16 @@ pub(super) fn get_unused_fields(
     let fields = match fields {
         syn::Fields::Unit => Vec::new(),
         syn::Fields::Unnamed(_) => Vec::new(),
-        syn::Fields::Named(fields) => fields
-            .named
-            .iter()
-            .filter(|field| {
-                field
-                    .ident
-                    .as_ref()
-                    .map_or(true, |idn| ignore.contains(&idn.to_string()))
-            })
-            .cloned()
-            .collect(),
+        syn::Fields::Named(fields) => fields.named.iter().cloned().collect(),
     };
-
     fields
         .iter()
         .filter(|&field| {
             // Safety: Named fields are guaranteed to have an identifier.
             #[allow(clippy::unwrap_used)]
             let field_name = format!("{}", field.ident.as_ref().unwrap());
-            !message.contains(&field_name)
+            !message.contains(&field_name) && !ignore.contains(&field_name)
```

Comment:
This should be either:

```rust
!(message.contains(&field_name) && ignore.contains(&field_name))
```

```rust
!message.contains(&field_name) || !ignore.contains(&field_name)
```

Reply:
let's consider the example, field `data` exists in message, then it should be `false`. If it doesn't exist in the message, then we check if it exists in ignore, if yes, then `false`. Which translates to `(!message.contains(&field_name))` this is the first condition, if this doesn't short circuit, then evaluate the second condition `(!ignore.contains(&field_name))`

This translates to:
either
```rust
!message.contains(&field_name) && !ignore.contains(&field_name)
```
or
```rust
!(message.contains(&field_name) || ignore.contains(&field_name))

Reply:
My bad, you're right.